src/lib/tuya.ts

```diff
@@ -1312,8 +1312,211 @@ const tuyaExtend = {
+                if (convertedValue === undefined) {
+                    // conversion done inside converter, ignore.
+                } else if (attribute.type == dataTypes.bool) {
+                    await sendDataPointBool(entity, attribute.dp, convertedValue as boolean, sendCommand, seq);
+                } else if (attribute.type == dataTypes.number) {
+                    await sendDataPointValue(entity, attribute.dp, convertedValue as number, sendCommand, seq);
+                } else if (attribute.type == dataTypes.string) {
+                    await sendDataPointStringBuffer(entity, attribute.dp, convertedValue as string, sendCommand, seq);
+                } else if (attribute.type == dataTypes.raw) {
+                    await sendDataPointRaw(entity, attribute.dp, convertedValue as number[], sendCommand, seq);
+                } else if (attribute.type == dataTypes.enum) {
+                    await sendDataPointEnum(entity, attribute.dp, convertedValue as number, sendCommand, seq);
+                } else if (attribute.type == dataTypes.bitmap) {
+                    await sendDataPointBitmap(entity, attribute.dp, convertedValue as number, sendCommand, seq);
+                } else {
+                    throw new Error(`Don't know how to send type '${typeof convertedValue}'`);
+                }
+
+                state[convertedKey] = value;
+            }
+            return {state};
+        },
+    }];
+
+    return {fromZigbee, toZigbee, isModernExtend: true};
+}
+
+export interface TuyaEnumLookupArgs {
+    name: string, attribute: {dp: number, type: number}, lookup: KeyValue,
+    description?: string, readOnly?: boolean, endpoint?: string, skip?: (meta: Tz.Meta) => boolean,
+    expose?: Expose,
+}
+export interface TuyaBinaryArgs {
+    name: string, attribute: {dp: number, type: number}, valueOn: [string | boolean, unknown], valueOff: [string | boolean, unknown],
+    description?: string, readOnly?: boolean, endpoint?: string, skip?: (meta: Tz.Meta) => boolean,
+    expose?: Expose,
+}
+
+export interface TuyaNumericArgs {
+    name: string, attribute: {dp: number, type: number},
+    description?: string, readOnly?: boolean, endpoint?: string, unit?: string, skip?: (meta: Tz.Meta) => boolean,
+    valueMin?: number, valueMax?: number, valueStep?: number, scale?: number | [number, number, number, number],
+    expose?: exposes.Numeric,
+}
+
+export interface TuyaLightArgs {
+    state: {dp: number, type: number, valueOn: [string | boolean, unknown], valueOff: [string | boolean, unknown], skip?: (meta: Tz.Meta) => boolean},
+    brightness: {dp: number, type: number, scale?: number | [number, number, number, number]},
+    max?: {dp: number, type: number, scale?: number | [number, number, number, number]},
+    min?: {dp: number, type: number, scale?: number | [number, number, number, number]},
+    endpoint?: string,
+}
+
+const tuyaModernExtend = {
+    enumLookup(args: TuyaEnumLookupArgs): ModernExtend {
```

Comment:
I would prefix all of these with `datapoint`, e.g. `datapointLight` because not all TuYa devices work via datapoints.

Reply:
maybe a shorter prefix? for example "dp". e.g. "dpLight", "dpEnumLookup" :)

Reply:
That's fine!